### ISSUE TITLE
Question: Is there a way to get an export from excalidraw to mermaid?

### ISSUE THREAD

**Issue by danieltome** [excalidraw/mermaid-to-excalidraw#66]
I've tried searching but can't find anything available. Do you know if there's a way to do excalidraw to mermaid?

**Comment by chiefkana** [comment]
Same thing here!

**Comment by ad1992** [comment]
No, currently there is no way to convert from Excalidraw to Mermaid. 
Could you please share the scenarios where you would need this feature?

**Comment by chiefkana** [comment]
Most of My Company's frontend documentation is based on drawings made in excalidraw, as of now we are migrating to customizable mermaid snippets in frontends wiki, so the question appeared. If it is not a case as of now, will be happy to work on it.

**Comment by DenFin** [comment]
This would be incredibly helpful. I'm interested in this as well but wouldn't know where to start honestly

**Comment by bselwayregis** [comment]
This is pretty much exactly my use case as well ☝️

**Comment by jpetrantoni** [comment]
+1 on this

I think theres too many edge cases for it to be officially supported. The best bet might to use the outputted json to reconstruct it.

**Comment by YokiToki** [comment]
In my opinion, the main problem is that arrows don't have a start and end binding to a generic element, sometimes they have a text binding, but that's not necessary. As a result, we end up with sets of linear, generic, and text elements unrelated to each other. The only point of reference is the positioning and size of the element.

**Comment by YokiToki** [comment]
Hi everyone! I have tried to implement excalidraw to mermaid conversion. This only works well for .excalidraw files that are created from .mermaid imports or manually properly customized arrows. For most cases it doesn't work for the moment. I'd be happy if anyone decides to contribute. https://github.com/YokiToki/excalidraw-to-mermaid

**Comment by dudido** [comment]
I came here looking for an answer to that question. Mermaid is natively supported across various platforms like Notion

**Comment by mjga-trifork** [comment]
Personally I have a similar use case. I like drawing stuff in Excalidraw, not so much in mermaid, but we use mermaid in github readmes so the diagrams are versioned. So initially drawing a diagram and exporting it to mermaid would be nice.

As a workaround one can feed the Excalidraw diagram to ChatGPT and ask it to convert it to a mermaid diagram. It's not perfect, but it gives you a starting point.

**Comment by Lunatix01** [comment]
> Personally I have a similar use case. I like drawing stuff in Excalidraw, not so much in mermaid, but we use mermaid in github readmes so the diagrams are versioned. So initially drawing a diagram and exporting it to mermaid would be nice.
> 
> As a workaround one can feed the Excalidraw diagram to ChatGPT and ask it to convert it to a mermaid diagram. It's not perfect, but it gives you a starting point.

I have a similar use case to yours, hopefully, we will see that feature.

**Comment by chuck00lin** [comment]
+1 on this, it will be nice feature!

**Comment by whitewatercn** [comment]
+1 on this

**Comment by alexsoaresok** [comment]
+1 on this 🙌

**Comment by nefagi-01** [comment]
+1

**Comment by ugur93** [comment]
+1

**Comment by edumaciel10** [comment]
+1

**Comment by CsarChvz** [comment]
+1

**Comment by kakkoyun** [comment]
+1

**Comment by SimonMaghiar** [comment]
+1

**Comment by HelloAlexPan** [comment]
+1

**Comment by nandastone** [comment]
+1

**Comment by visuman** [comment]
+1

**Comment by kamal-hr** [comment]
+1

**Comment by ddpolyakov** [comment]
+1

**Comment by Merieli** [comment]
let's think about an MVP initially supporting only the mermaid of the flowchart type? That would already be wonderful

**Comment by amsteel** [comment]
+1 on this
👍

**Comment by toransahu** [comment]
+1

**Comment by andreskimlee** [comment]
+1

**Comment by TejuAdey** [comment]
+1000.

**Comment by markdall** [comment]
This would close the loop, letting us go from Excalidraw->Mermaid->Excalidraw while storing revisions of diagrams in git. That would be pure bliss.

**Comment by benpriestman** [comment]
👍 from me

**Comment by alexzadeh** [comment]
+1 for this, will be huge for iterating with LLMs!

**Comment by NanmiCoder** [comment]
+1 for this, will be huge for iterating with LLMs!

**Comment by bansalvks** [comment]
+9000

**Comment by YouKyi** [comment]
+1

**Comment by yeo-gen** [comment]
+1

**Comment by sindrel** [comment]
Hi all, I’m the developer of an open-source CLI tool called Excalidraw Converter. It started as a way to convert Excalidraw diagrams to Gliffy and draw.io formats, but I just added support for Mermaid flowcharts (a feature I needed myself).

I came across this thread while researching, and thought this might be relevant. The tool is cross-platform, free, and open-source—just something I built to solve my own workflow needs. If it’s useful to anyone else here, that’s great.

You can find it here: https://github.com/sindrel/excalidraw-converter

**Comment by ChevTech** [comment]
+1

**Comment by ruben-davia** [comment]
+1, did anyone find a solution yet? Even a partial one?

**Comment by gaddlord** [comment]
ChatGPT/Gemini/etc. come to the rescue. I had converted a lot of diagrams that way.

<img width="804" height="763" alt="Image" src="https://github.com/user-attachments/assets/dd20b284-1136-461a-8964-c2cc0a266b42" />

**Comment by moona3k** [comment]
Here's a tool that converts does this - https://github.com/moona3k/excalidraw-to-mermaid

```
npx excalidraw-to-mermaid diagram.excalidraw
```

It handles rectangles, diamonds, ellipses, rounded corners, arrow styles (solid/dashed/thick), edge labels, subgraphs, and auto-detects flow direction.

**Comment by toransahu** [comment]
Thanks for the tool @moona3k
